src/main/java/com/xiaomi/infra/pegasus/rpc/async/ReplicaSession.java

```diff
@@ -326,6 +326,10 @@ void tryNotifyFailureWithSeqID(int seqID, error_types errno, boolean isTimeoutTa
   }
 
   private void write(final RequestEntry entry, VolatileFields cache) {
+    if (!interceptorManager.onSendMessage(this, entry)) {
+      return;
```

Comment:
before `write` seem no need add `interceptor`, you can put `intercepter` into `markSessionConnected` such as https://github.com/XiaoMi/pegasus-java-client/blob/3615cfd978797334cb7beec4b624473642698a63/src/main/java/com/xiaomi/infra/pegasus/rpc/async/ReplicaSession.java#L221

It's also means 
```java
isNegotiationRequest(entry) || !session.tryPendRequest(entry);
```
can be put into `onConnected`

Reply:
No, we should call this interceptor when a message is sending, which means it will be called twice times if you send two messages. But `onConnected` is called only when the session is connected, and it will only called once.